src/main/scala/battery/bot/core/CommandProcess.scala

```diff
@@ -114,6 +115,57 @@ class CommandProcess(persistenceService: PersistenceService, telegramClient: Tel
     }
   }
 
+  def userDevices(result: Update): IO[Unit] = {
+    for {
+      userId      <- persistenceService.getUserID(result.message.from.username)
+      devicesList <- persistenceService.getUserDevicesName(userId)
+      _           <- updateDevice(result, devicesList)
+    } yield ()
+  }
+
+  def updateDevice(result: Update, devicesList: List[String]): IO[Unit] = {
+    val data = result.message.text.split(";").toList
+    data match {
+      case _ :: name :: chargingTime :: Nil if chargingTime.toDoubleOption.nonEmpty && devicesList.contains(name) =>
```

Comment:
because the first list item is always ignored what doing `.tail` on the list before match